### ISSUE TITLE
some questions about Paper "Guided Collaborative Training"

### ISSUE THREAD

**Issue by czy341181** [ZHKKKe/PixelSSL#7]
great work. Thanks for your amazing codebase.
I have some questions about this paper "Guided Collaborative Training for Pixel-wise
Semi-Supervised Learning"

1.I'm wondering if I can just use max score of a pixel as an evaluation criterion without Flaw Detector in **semantic segmentation task**?  If so, how would it work if I use score directly, have you ever done such experiment?

2. Is Flaw Correction Constraint forcing the error to 0 to correct the result of semantic segmentation?  This loss, not quite understand what it means.

**Comment by ZHKKKe** [comment]
Hi, thanks for your attention.

For your questions:
**Q1: I'm wondering if I can just use max score of a pixel as an evaluation criterion without Flaw Detector in semantic segmentation task? If so, how would it work if I use score directly, have you ever done such experiment?**
Sorry that we were not try it since we we also considered the regression tasks ("max score" is unavailable in the regression tasks).

**Q2: Is Flaw Correction Constraint forcing the error to 0 to correct the result of semantic segmentation? This loss, not quite understand what it means.**
Yes,  Flaw Correction Constraint can correct the result of semantic segmentation. 
For this point, you can check Sec 3.4 in our paper. We tried to compare Flaw Correction Constraint with the GAN loss to analyze its ability. Please feel free to ask more detailed questions about this.

**Comment by czy341181** [comment]
about Q1. For a pixel-level classification task, 
 compared to Flaw detectors if you use the classification score directly，What is your opinion and what will be the effect. 
I think it is clearer and more naive to use the segmentation score directly.

**Comment by ZHKKKe** [comment]
Yes, I agree with you. Using the classification score in semantic segmentation is more straightforward.
In GCT, we do not use such a solution because we want to generalize it to the regression tasks.

**Comment by czy341181** [comment]
Thanks for your patient reply! Best

**Comment by czy341181** [comment]
Hi, sorry to bother you.  I have a question about codebase in GCT.

In this paper, two networks have different initializations to form the perturbations between them. But I can't find where are the different initialization settings.

`        self.l_model = func.create_model(model_funcs[0], 'l_model', args=self.args)
        self.r_model = func.create_model(model_funcs[1], 'r_model', args=self.args)`

`    model = mclass(**kwargs)
    model = torch.nn.DataParallel(model)
    model = model.cuda()
    
    logger.log_info('  ' + '=' * 76 + '\n  {0} parameters \n{1}'.format(mname, model_str(model)))
    return model`

Could you tell me where is it?

**Comment by ZHKKKe** [comment]
You can find the random initailization step in the model defination. For example: https://github.com/ZHKKKe/PixelSSL/blob/ce192034355ae6a77e47d2983d9c9242df60802a/task/sseg/module/deeplab_v2.py#L78-L79

Initailization usually done by the nn.Module class, not in the SSL framework. 
Thanks.

**Comment by czy341181** [comment]
So both models are randomly initialized to ensure to be different?

**Comment by ZHKKKe** [comment]
Yes, exactly.
